tests/xharness/UnsupportedOperationException.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Runtime.Serialization;
+
+namespace Xharness {
+	public class UnsupportedOperationException : Exception {
```

Comment:
Sounds like this is the same as `InvalidOperationException`, could we use that exception type instead of creating a new one?

Reply:
I am using it in tests to distinguish that proper behavior happened. With `InvalidOperationException`  I was worried some other either ours or not even ours code would throw the same. I also wasn't going to base the unit test on comparing exception message. What do you think?

Reply:
That seems like a rather remote possibility, that you'd get `InvalidOperationException` for a different cause than the expected one. I still think it's best to not create a new exception type, because the best practices for exceptions say ["Introduce a new exception class only when a predefined one doesn't apply"](https://docs.microsoft.com/en-us/dotnet/standard/exceptions/best-practices-for-exceptions#use-the-predefined-net-exception-types).

Reply:
I reverted to `InvalidOperationException`